pkg/endpoints/websocket_test.go

```diff
@@ -10,526 +10,435 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-	awaitFatal(func() bool { return atomic.LoadInt32(&TaskRunCreationsRecorded) == int32(clients) },
-		fmt.Sprintf("Expected %d CreationsRecorded for TaskRuns", clients))
-
-	// Update taskrun and wait until clients receive
-	r.updateTestTaskRun("WebsocketTaskrun", "65432")
-	awaitFatal(func() bool { return atomic.LoadInt32(&TaskRunUpdatesRecorded) == int32(clients) },
-		fmt.Sprintf("Expected %d UpdatesRecorded for TaskRuns", clients))
-
-	// Delete taskrun and wait until clients receive
-	r.deleteTestTaskRun("WebsocketTaskrun")
-	awaitFatal(func() bool { return atomic.LoadInt32(&TaskRunDeletionsRecorded) == int32(clients) },
-		fmt.Sprintf("Expected %d DeletionsRecorded for TaskRuns", clients))
-
-	// PipelineResources
-	// Create pipelineresource and wait until clients receive
-	r.createTestPipelineResource("WebsocketPipelineResource", "12345")
-	awaitFatal(func() bool { return atomic.LoadInt32(&PipelineResourceCreationsRecorded) == int32(clients) },
-		fmt.Sprintf("Expected %d CreationsRecorded for PipelineResources", clients))
-
-	// Update pipelineresource and wait until clients receive
-	r.updateTestPipelineResource("WebsocketPipelineResource", "65432")
-	awaitFatal(func() bool { return atomic.LoadInt32(&PipelineResourceUpdatesRecorded) == int32(clients) },
-		fmt.Sprintf("Expected %d UpdatesRecorded for PipelineResources", clients))
-
-	// Delete pipelineresource and wait until clients receive
-	r.deleteTestPipelineResource("WebsocketPipelineResource")
-	awaitFatal(func() bool { return atomic.LoadInt32(&PipelineResourceDeletionsRecorded) == int32(clients) },
-		fmt.Sprintf("Expected %d DeletionsRecorded for PipelineResources", clients))
-
-	T.Log("Waiting for clients to terminate...")
+	awaitFatal(awaitAllClients, t, fmt.Sprintf("Expected %d clients within pool", clients))
+
+	// Create, Update, and Delete records
+	CUDPipelineResources(r, t, namespace)
+	CUDPipelines(r, t, namespace)
+	CUDPipelineRuns(r, t, namespace)
+	CUDTasks(r, t, namespace)
+	CUDTaskRuns(r, t, namespace)
+	// Create and Delete records
+	CDNamespaces(r, t)
+
+	t.Log("Waiting for clients to terminate...")
 	wg.Wait()
 
-	awaitFatal(func() bool { return resourcesBroadcaster.PoolSize() == 0 },
-		"runsBroadcaster pool should be empty")
-
-	T.Log("Exit ResourcesRunWebsocket")
+	awaitNoClients := func() bool {
+		return ResourcesBroadcaster.PoolSize() == 0
+	}
+	awaitFatal(awaitNoClients, t, "Pool should be empty")
+
+	// Check that all fields have been populated
+	for _, informerRecord := range recordMap {
+		creates := int(informerRecord.Create())
+		updates := int(informerRecord.Update())
+		deletes := int(informerRecord.Delete())
+		// records without an update hook
+		if updates == -1 {
+			if creates != clients && creates != deletes {
+				t.Fatalf("CD informer %s creates[%d] and deletes[%d] not equal expected to value: %d\n", informerRecord.CRD, creates, deletes, clients)
+			}
+		} else {
+			if creates != clients && creates != deletes && creates != updates {
```

Comment:
Talked to Vincent about this. Should have `creates > 0`

Reply:
I recognize this should be || rather than && as well so I will clean this up.